File: Lab1/Task2Java/app/src/main/java/com/example/musicplayerapp/MainActivity.java
package com.example.musicplayerapp;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import com.example.musicplayerapp.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.startMusicButton.setOnClickListener(view -> {
            Intent intent = new Intent(this, MusicService.class);
            startService(intent); // Start the foreground service
        });

        binding.stopMusicButton.setOnClickListener(view -> {
            Intent intent = new Intent(this, MusicService.class);
            stopService(intent); // Stop the foreground service
        });
    }
}

File: Lab1/Task2Kotlin/app/src/main/java/com/example/musicplayerapp/MainActivity.kt
package com.example.musicplayerapp

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.musicplayerapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.startMusicButton.setOnClickListener {
            val intent = Intent(this, MusicService::class.java)
            startService(intent) // Start the foreground service
        }

        binding.stopMusicButton.setOnClickListener {
            val intent = Intent(this, MusicService::class.java)
            stopService(intent) // Stop the foreground service
        }
    }
}
